### DISCUSSION TITLE
Export Compile Command Issue | Can't Find Headers

### DISCUSSION THREAD

**Discussion by utsur0** [ArthurSonzogni/FTXUI#387] [Q&A] [answered]
I'm a bit new to CMake and have been following the Introduction Documentation utilizing the example `CMakeLists.txt` with its preamble to Fetch this project. 

However when adding the option `set(CMAKE_EXPORT_COMPILE_COMMANDS ON)` inside `CMakeLists.txt` the resulting `compile_commands.json` can't find the .hpp files inside `./_deps/ftxui-src/include` using clang. This occurs despite the dir being included. I've been researching a solution for hours (maybe I just don't know what I'm looking for 😞 ) my apologies if this is a stupid issue. 

SysInfo:
macOS: `12.0.1`
clang version: `13.0.1` (Homebrew | Target: arm64-apple-darwin21.1.0)
cmake version: `3.23.1`

<details>
  <summary>CMakeLists.txt</summary>
Haven't changed much aside from the tag and adding the flag. Also my main.cpp is an example from the documentation.
  
 ```
cmake_minimum_required (VERSION 3.11)
 
# --- Fetch FTXUI --------------------------------------------------------------
include(FetchContent)
 
set(FETCHCONTENT_UPDATES_DISCONNECTED TRUE)
FetchContent_Declare(ftxui
  GIT_REPOSITORY https://github.com/ArthurSonzogni/ftxui
  GIT_TAG v2.0.0
)
 
FetchContent_GetProperties(ftxui)
if(NOT ftxui_POPULATED)
  FetchContent_Populate(ftxui)
  add_subdirectory(${ftxui_SOURCE_DIR} ${ftxui_BINARY_DIR} EXCLUDE_FROM_ALL)
endif()
 
# ------------------------------------------------------------------------------
 
project(tui
  LANGUAGES CXX
  VERSION 1.0.0
)

set(CMAKE_EXPORT_COMPILE_COMMANDS ON) 
add_executable(tui src/main.cpp)
target_include_directories(tui PRIVATE src)
 
target_link_libraries(tui
  PRIVATE ftxui::screen
  PRIVATE ftxui::dom
  PRIVATE ftxui::component # Not needed for this example.
)
  ```
</details>

<details>
  <summary>compile_commands.json</summary>
  
  ```
[
{
  "directory": "/Volumes/OS/Volumes/OS/Users/winter/Development/TUI",
  "command": "/Library/Developer/CommandLineTools/usr/bin/c++  -I/Volumes/OS/Users/winter/Development/TUI/src -I/Volumes/OS/Volumes/OS/Users/winter/Development/TUI/_deps/ftxui-src/include  -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.0 -std=gnu++17 -o CMakeFiles/tui.dir/src/main.cpp.o -c /Volumes/OS/Volumes/OS/Users/winter/Development/TUI/src/main.cpp",
  "file": "/Volumes/OS/Volumes/OS/Users/winter/Development/TUI/src/main.cpp"
}
]
  ```
  
</details>

Is it because of firmlinks on macOS? CMake issue? Clang Issue? Did I configure something wrong? _Think I've been chasing rabbits._

If anymore information is needed please let me know and I'll update!   Thanks!

**Comment by ArthurSonzogni** [comment, marked as answer]
I tried your CMakeLists.txt on Linux: I was able to compile and execute "tui".

I got compile_commands.json similar to yours:
```json
[
{
  "directory": "/tmp/test/build",
  "command": "/usr/bin/c++   -I/tmp/test/src -I/tmp/test/build/_deps/ftxui-src/include    -std=gnu++17 -o CMakeFiles/tui.dir/src/main.cpp.o -c /tmp/test/src/main.cpp",
  "file": "/tmp/test/src/main.cpp"
}
]

```

If I execute the command from an empty directory, I get:
```
/usr/bin/c++   -I/tmp/test/src -I/tmp/test/build/_deps/ftxui-src/include    -std=gnu++17 -o CMakeFiles/tui.dir/src/main.cpp.o -c /tmp/test/src/main.cpp
/tmp/test/src/main.cpp:3:10: fatal error: ftxui/dom/elements.hpp: Aucun fichier ou dossier de ce type
    3 | #include "ftxui/dom/elements.hpp"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

However, if I execute the command from a directory, where "cmake" has been called, the commands works.
It is because the "_deps/ftxui-src" directory is fetched by cmake.

Does this resolve your issue?

**Comment by utsur0** [comment]
Oddly I _was_ executing it from the dir I ran `cmake` in. I'll mark this a resolved because clang is finicky with its resolutions. I hate to admit—resetting the PC fixed my issue. Gotta assume clang was grabbing the wrong compile_commands.json despite me explicitly setting it within my IDE. Why? I'm not too sure—probably an issue with that program.

Anyways I deleted my other configured .json's, restarted, and then it worked.
Doubt this will be a problem or solution for many others so...my apologies.
